src/SideNavigation/SideNavigation.js

```diff
@@ -1,209 +1,276 @@
+                    {sideNavList}
+                </div>
+            );
+        }
 
-                obj[id] = false;
-                initialState.push(obj);
-            }
-        });
+        return sideNavList;
+    }
+}
+
+SideNavList.propTypes = {
+    children: PropTypes.node,
+    className: PropTypes.string,
+    hasParent: PropTypes.bool,
+    open: PropTypes.bool,
+    selectedId: PropTypes.string,
+    title: PropTypes.string,
+    titleProps: PropTypes.object,
+    onItemSelect: PropTypes.func
+};
+
+SideNavList.propDescriptions = {
+    hasParent: 'Internal use only',
+    open: 'Set to **true** to mark `SideNavSubList` as open.',
+    selectedId: 'Internal use only',
+    onItemSelect: 'Internal use only'
+};
+
+SideNavList.displayName = 'SideNavList';
+
+export class SideNavListItem extends React.Component {
+    constructor(props) {
+        super(props);
 
         this.state = {
-            selectedItem: 'item_2',
-            itemStates: initialState
+            expanded: this.props.expanded ? this.props.expanded : false
         };
     }
 
-    handleSelectChild = (e, id) => {
+    getDerrivedStateFromProps(updatedProps, previousState) {
+        if (updatedProps.expanded !== previousState.expanded) {
+            return { expanded: updatedProps.expanded };
+        }
+    }
+
+    handleExpand = () => {
         this.setState({
-            selectedItem: id
+            expanded: !this.state.expanded
         });
     };
 
-    handleSelect = (e, id) => {
-        let iStates = Object.assign({}, this.state.itemStates);
-        iStates[id] = !iStates[id];
-        this.setState({ itemStates: iStates });
-        this.setState({ selectedItem: id });
-    };
+    render() {
+        const { children, glyph, id, isSubItem, name, onClick, onItemSelect, selected, selectedId, url, ...props } = this.props;
+        const getClasses = () => {
+            return classnames(
+                {
+                    'fd-side-nav__link': !isSubItem,
+                    'fd-side-nav__sublink': isSubItem,
+                    'is-selected': selected,
+                    'has-child': hasChild,
+                    'is-expanded': this.state.expanded
+                }
+            );
+        };
 
-    getLinkClasses = ({ id, hasChild }) => {
-        return classnames(
-            'fd-side-nav__link',
-            {
-                'is-selected': this.state.selectedItem === id,
-                'has-child': hasChild,
-                'is-expanded': this.state.itemStates[id] && hasChild
-            }
-        );
-    }
+        const renderLink = () => {
+            return (
+                <a
+                    className={getClasses()}
+                    href={url}
+                    onClick={(e) => {
+                        onClick(e);
+                        onItemSelect(e, id, hasChild);
+                        if (hasChild) {
+                            this.handleExpand();
+                        }
```

Comment:
I found what I _think_ is a bug, but wanted to get thoughts on it first before creating an issue.  It would be a follow-up task if it was a bug since the behavior existed previously.  When click on a nav item that is the expand/collapse item, it changes the `selectedId`, but these items themselves shouldn't actually be "selectable".  They are simply used to expand and collapse the children within it.

Reply:
I agree this was an issue. Pushed 33e7b51 to fix this issue and 0935d11 to fix the corresponding test.